### PR TITLE
Add TS type definitions

### PR DESCRIPTION
**Pull request by ricardo-devis-agullo**
Add TypeScript definition file.

Closses #12 

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows [Code of conduct](https://github.com/fastify/fastify/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Would you mind adding a unit test? Check out https://github.com/fastify/fastify-static for an example.

**Comment by ricardo-devis-agullo** [comment]
For sure, done! 

To follow the same structure as that one, I also moved the rest of the unit tests to a `test/` folder :)

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Definitions by: ricardo-devis-agullo <https://github.com/ricardo-devis-agullo>
```
Comment (Eomm) [inline index.d.ts]: as this https://github.com/fastify/fastify-bearer-auth/pull/29/files#r336102699 feedback I would remove this.

Your contribution will remain in github and `git blame` forever 💪🏻😃
If you found it in other plugins it is an oversight

**Comment by nboltn** [comment]
why is this still open?

**Comment by mcollina** [comment]
@Eomm I'll drop that line in a commit on master.

**Pull request merged**